# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/locks/LockManager.cpp
# Commit: 588d5d470544
# Full Hash: 588d5d4705444710c4a6c35290fc92d802557776
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2023-03-11 09:54:30
# Description:
#   Bug 1819146, don't try to use destroyed global when accessing signal's reason, r=peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D172131
# ==============================================================================

diff -r 6ac0d009f1ef -r 588d5d470544 dom/locks/LockManager.cpp
--- a/dom/locks/LockManager.cpp	Fri Mar 10 15:00:39 2023 +0000
+++ b/dom/locks/LockManager.cpp	Fri Mar 10 15:14:32 2023 +0000
@@ -97,9 +97,13 @@
       return false;
     }
     if (options.mSignal.Value().Aborted()) {
-      AutoEntryScript aes(options.mSignal.Value().GetParentObject(),
-                          "LockManager::Request");
-      JSContext* cx = aes.cx();
+      AutoJSAPI jsapi;
+      if (!jsapi.Init(options.mSignal.Value().GetParentObject())) {
+        aRv.ThrowNotSupportedError("Signal's realm isn't active anymore.");
+        return false;
+      }
+
+      JSContext* cx = jsapi.cx();
       JS::Rooted<JS::Value> reason(cx);
       options.mSignal.Value().GetReason(cx, &reason);
       aRv.MightThrowJSException();